elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/Schema.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation;
+
+import com.yahoo.elide.core.DataStore;
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.datastores.aggregation.annotation.Cardinality;
+import com.yahoo.elide.datastores.aggregation.annotation.DimensionSize;
+import com.yahoo.elide.datastores.aggregation.annotation.FriendlyName;
+import com.yahoo.elide.datastores.aggregation.annotation.MetricAggregation;
+import com.yahoo.elide.datastores.aggregation.annotation.MetricComputation;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.annotation.Annotation;
+import java.util.AbstractMap;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.persistence.Temporal;
+
+/**
+ * {@link Schema} keeps track of table, metrics, and dimensions of an entity for AggregationDataStore.
+ * <p>
+ * On calling {@link DataStore#populateEntityDictionary(EntityDictionary)}, one {@link Schema} will be created for each
+ * entity. AggregationDataStore will use {@link Schema} for each entity to determine whether
+ * <ul>
+ *     <li> an attribute is a metric
+ *     <li> an attribute or relationship is a dimension
+ *     <li> an attribute is a Data/Time column and, thus, supports grain
+ *     <li> an Attribute is a friendlyName or human displayable column for that entity
+ *     <li> a dimension has {@link DimensionSize#SMALL}, {@link DimensionSize#MEDIUM}, or {@link DimensionSize#LARGE}
+ *          cardinality
+ * </ul>
+ * <p>
+ * {@link Schema} is thread-safe and can be accessed by multiple-threads.
+ */
+@Slf4j
+public class Schema {
+
+    @Getter
+    private final Class<?> entityClass;
+
+    @Getter
+    private final String dateTimeColumn;
+
+    private final String friendlyName;
+
+    @Getter(value = AccessLevel.PRIVATE)
+    private final List<String> metrics;
+
+    @Getter(value = AccessLevel.PRIVATE)
+    private final List<String> dimensions;
+
+    @Getter(value = AccessLevel.PRIVATE)
+    private final Map<String, DimensionSize> dimensionSizes;
+
+    @Getter(value = AccessLevel.PRIVATE)
+    private final EntityDictionary entityDictionary;
+
+    /**
+     * Constructor.
+     *
+     * @param clz  The type of the entity, whose {@link Schema} is to be constructed
+     * @param entityDictionary  The meta info object that helps to construct {@link Schema}
+     */
+    public Schema(Class<?> clz, EntityDictionary entityDictionary) {
```

Comment:
Can we change "clz" to "entityClass"?

Reply:
I'm following the codebase convention https://github.com/yahoo/elide/search?q=cls&unscoped_q=cls